fuzzer/src/main.rs

```diff
@@ -1,78 +1,216 @@
 use std::collections::{BTreeMap, BTreeSet, VecDeque, LinkedList, BinaryHeap};
 use std::time::Duration;
 
-use bitvec::{vec::BitVec, order::Msb0};
+use bitvec::{vec::BitVec, order::Msb0, order::BitOrder, store::BitStore};
 use honggfuzz::fuzz;
 use parity_scale_codec::{Encode, Decode, Compact};
+use honggfuzz::arbitrary::{Arbitrary, Unstructured, Result as ArbResult};
 
-#[derive(Encode, Decode)]
+#[derive(Encode, Decode, Clone, PartialEq, Debug, Arbitrary)]
 pub struct MockStruct{
 	vec_u: Vec<u8>
 }
 
-#[derive(Encode, Decode)]
+#[derive(Encode, Decode, Clone, PartialEq, Debug)]
+/// Used for implementing the Arbitrary trait for a BitVec.
+pub struct BitVecWrapper<O: BitOrder, T: BitStore>(BitVec<O, T>);
+
+impl<O: 'static + BitOrder, T: 'static + BitStore + Arbitrary> Arbitrary for BitVecWrapper<O, T> {
+	fn arbitrary(u: &mut Unstructured<'_>) -> ArbResult<Self> {
+		let v = Vec::<T>::arbitrary(u)?;
+		Ok(BitVecWrapper(BitVec::<O, T>::from_vec(v)))
+	}
+}
+
+
+#[derive(Encode, Decode, Debug, Clone, Arbitrary)]
+/// Used for implementing the PartialEq trait for a BinaryHeap.
+struct BinaryHeapWrapper(BinaryHeap<u32>);
+
+impl PartialEq for BinaryHeapWrapper {
+	// Eq here is implemented by creating a vector of all values and then sorting it.
+	// TODO: A better approach would be to use `into_iter_sorted` and compare both iterator, once
+	// https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.into_iter_sorted is stable.
```

Comment:
This is a test and nobody cares how fast tests are^^
```suggestion
```

Reply:
So given this test is used for fuzzing purposes, one could argue that actually the rate at which we can compare two binaryheap will directly impact the fuzzing speed. But maybe I'm missing something here :)

Reply:
While you are right that the fuzzing rate could be improved, I don't expect any improvements over the solution I have posted below. We would still need to `clone()` and after briefly looking over the implementation, it seems that `into_iter_sorted` is essentially doing the same as `into_sorted_vec`. You could argue that you can stop earlier with an iterator, but yeah. I don't know. If we ever run this fuzzer and get into troubles we can improve.

Reply:
Yup I guess the improvement would be minor. I had not seen `into_iter_sorted` when writing the code, so what you suggested is definitely better than my implementation. Thanks! :)